### DISCUSSION TITLE
Clarification on BoundedCapacity for Dataflow

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/runtime#66878] [Q&A] [answered]
How does ```BoundedCapacity``` control the processing of data?

If I have the pipeline:
```A -> B -> C```
And ```B``` has a bounded capacity of 10, does that mean:
1) B can have up to 10 items waiting for it to process.
2) C can have up to 10 items waiting for it to process.

Does it control an input buffer or output buffer?

**Comment by gfoidl** [comment, marked as answer]
> And `B` has a bounded capacity of 10, does that mean:

It means that `B` can buffer max. 10 items.
So `A` can't write to `B` successfully if `B`s buffer is full, so `A` gets backpressure from `B`.

Although the concept of input and output buffer doesn't explicitely exist, this one refers to input (from `A` to `B`, see above).

Best is to create a simple console app and play around with different producer and consumer speed of `A`, `B`, and `C` to see whats going on (with debugger attached / or console output).
